gitlabform/processors/project/schedules_processor.py

```diff
@@ -42,102 +48,124 @@ def _process_configuration(self, project_and_group: str, configuration: dict):
-        created_schedule = self.gitlab.create_pipeline_schedule(
-            project_and_group,
-            schedule_description,
-            data.get("ref"),
-            data.get("cron"),
-            optional_data=data,
-        )
-        self.__set_schedule_variables(
-            project_and_group,
-            created_schedule.get("id"),
-            configuration.get("schedules|" + schedule_description + "|variables"),
+        data = configured_schedules[schedule_description]
+
+        schedule_data = {"description": schedule_description, **data}
+        debug("Creating pipeline schedule using data: '%s'", schedule_data)
+
+        created_schedule_id = project.pipelineschedules.create(schedule_data).id
+        created_schedule = project.pipelineschedules.get(created_schedule_id)
+
+        self._set_schedule_variables(
+            created_schedule,
+            data.get("variables"),
         )
 
-    def __set_schedule_variables(self, project_and_group, schedule_id, variables):
-        schedule = self.gitlab.get_pipeline_schedule(project_and_group, schedule_id)
+        created_schedule.save()
 
-        existing_variables = schedule.get("variables")
+    @staticmethod
+    def _set_schedule_variables(schedule: ProjectPipelineSchedule, variables):
+        attributes = schedule.attributes
+        existing_variables = attributes.get("variables")
         if existing_variables:
-            debug(
-                "Deleting variables for pipeline schedule '%s'", schedule["description"]
-            )
+            debug("Deleting variables for pipeline schedule '%s'", schedule.description)
 
             for variable in existing_variables:
-                self.gitlab.delete_pipeline_schedule_variable(
-                    project_and_group, schedule_id, variable.get("key")
-                )
+                schedule.variables.delete(variable.get("key"))
 
         if variables:
             for variable_key, variable_data in variables.items():
-                self.gitlab.create_pipeline_schedule_variable(
-                    project_and_group,
-                    schedule_id,
-                    variable_key,
-                    variable_data.get("value"),
-                    variable_data,
-                )
+                variable_type = variable_data.get("variable_type")
+                if variable_type:
+                    schedule.variables.create(
+                        {
+                            "key": variable_key,
+                            "value": variable_data.get("value"),
+                            "variable_type": variable_data.get("variable_type"),
+                        }
+                    )
+                else:
+                    schedule.variables.create(
+                        {
+                            "key": variable_key,
+                            "value": variable_data.get("value"),
+                        }
+                    )
```

Comment:
Do you think we can just pass the entire `variable_data` to `schedule.variables.create`? Just wondering if that's doable and if it is, maybe it'll simplify the logic here.